### ISSUE TITLE
[RyuJIT/arm32][JitStress=1] Assertion failed '((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume)'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14542]
From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1".

Happens with `COMPlus_JitStress=1` (or `COMPlus_JitStress=2`).

Found in System.Drawing.Primitives.Tests:
```
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 13:12:47.45

E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1
E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_JitStress=1

E:\gh\corefx\bin\tests\System.Drawing.Primitives.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Drawing.Primitives.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Drawing.Primitives.Tests
Discovered:  System.Drawing.Primitives.Tests
Starting:    System.Drawing.Primitives.Tests

Assert failure(PID 22104 [0x00005658], Thread: 22980 [0x59c4]): Assertion failed '((consume == 0) && (produce == 0)) || (ComputeAvailableSrcCount(tree) == consume)' in 'System.Drawing.PrimitivesTests.SizeTests:DivideTestSizeFloat(int,float):this' (IL size 40)

    File: e:\gh\coreclr2\src\jit\lsra.cpp Line: 3726
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStress=1
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib @CarolEidt

**Comment by YongseopKim** [comment]
On linux/arm32,
```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10241532.log
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStress=1
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay

Using as the test runtime folder.
Running tests... Start time: 15:32:56
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet xunit.console.netcore.exe System.Drawing.Primitives.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Drawing.Primitives.Tests
Discovered:  System.Drawing.Primitives.Tests
Starting:    System.Drawing.Primitives.Tests

Assert failure(PID 16501 [0x00004075], Thread: 16501 [0x4075]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Reflection.Emit.DynamicMethod:GetDynamicMethodsModule():ref' (IL size 199)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/flowgraph.cpp Line: 21785
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet


Assert failure(PID 16501 [0x00004075], Thread: 16501 [0x4075]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Reflection.Emit.DynamicMethod:GetDynamicMethodsModule():ref' (IL size 199)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/flowgraph.cpp Line: 21785
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/overlay/dotnet

./RunTests.sh: line 86: 16501 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Drawing.Primitives.Tests.dll -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10241504/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=15:33:35. Return value was 134
Unable to find executable corerun
Copying core file core to /tmp/coredumps
error: One or more tests failed while running tests from ''.  Exit code 134.
1 test(s) failed
```
It is also affected by https://github.com/dotnet/coreclr/issues/14567 like #14549 .

```
 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.CLR 
2207dadf19fc976c6ccd04e12d46924a7b733457 Report registers as dead in GCInfo before the RhpPInvoke helper. (#14664)
5d66791a3a1932d2afcd3ef58dd0df91323662cb profiler changes for tiered compilation (#14612)
f8d6405c8e0680a1184dc07123aeb4abe62e9e87 Merge pull request #14647 from BruceForstall/AddArm64FrameLayoutDoc
67a66156d0307b49522a13618522d289268bc1f9 Fix for #12609 - add option to disable tiered compilation for profilers  (#14643)
33159233fd161877ffe37187bf06689fe0e8bea2 Fix RID extraction in packages and test build for Alpine (#14656)
458c90a363b22fb29f6581610d5349bf06e17971 Merge pull request #14649 from mikedn/float-absneg
2ba74317c027eb4ff1e854de02fbb9378d00a34e [Arm64] SIMD simple defines (#14628)
c0aa11ff835222facec5401c4074e5b46cb1635f [Arm64] SIMD lsra (#14631)
e9437e8a686d4db2a6667eb7f386279678916467 [Arm64] SIMD ins_Load (#14636)
4d253390b02d51dd5dc2049dbe83c122c3acba9d Merge pull request #14528 from jashook/pri1_rework

 dragon@loki  ~/dotnet/arm32-dotnet/tmp  cat coreoverlay-ARM-CHE-REL-10241504/COMMITS.FX 
7e91c485254e054e0f3c14291f97e6a382659bec Fix condition in System.Diagnostics.PerformanceCounter (#24828)
bcc59b2e74090cfbcc4d6e7c245c172744e9d46b Fix System.Data.Odbc configurations for support package and run UpdateVSConfigurations (#24752)
389d7ee0630e71ff868b4d9ebcdcf8111d5acee3 Merge pull request #24684 from Anipik/newbr
5d164eb843b6fc2278c107184634396642483901 Re-enabling previously disabled tests due to Retain bug (#24785)
1eabfe92ef9fe97f8abfccb044fd6bcfffa5bb5f Add some System.Collections.Concurrent perf tests (#24810)
a0c65af98df8fe746bb49da90b6c2a152d747a18 Remove New APIs in Syndication. (#24814)
d33e40e27e40d313977d324ae89c33f2842ee5b7 Deleted dead code from System.Net.Requests. (#24815)
c92789464b2ac983eba3a6021b13ef07eb0c8d8d Extentionclass added
725495577f7e17f1ae2ad2439f219203b2cef844 Add perf test to measure Memory.Span.Slice vs. Memory.Slice.Span (#24780)
90edf7d04a1d32266c275aa5b239b6022ebab98b Change Memory.Slice(...).Span to Memory.Span.Slice(...) (#24813)
```

**Comment by YongseopKim** [comment]
With d780f9a1327344bfc8e06737497b703cdbfba804, the tc passes on linux/arm32.

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10251014.log
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStress=1
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay

Using as the test runtime folder.
Running tests... Start time: 10:14:07
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay/dotnet xunit.console.netcore.exe System.Drawing.Primitives.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Drawing.Primitives.Tests
Discovered:  System.Drawing.Primitives.Tests
Starting:    System.Drawing.Primitives.Tests
Finished:    System.Drawing.Primitives.Tests

=== TEST EXECUTION SUMMARY ===
   System.Drawing.Primitives.Tests  Total: 2046, Errors: 0, Failed: 0, Skipped: 0, Time: 18.140s
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Drawing.Primitives.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=10:14:52. Return value was 0
All tests passed.

 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat ./tmp/coreoverlay-ARM-CHE-REL-10251012/COMMITS.CLR 
d780f9a1327344bfc8e06737497b703cdbfba804 fix lvaStressLclFldCB (#14676)
f324495d90ebe7cba7e10261a53cd82383bb317e Merge pull request #14658 from mikedn/legacy-reg-track
f09dd7ca32ff62dc25db9a0de0e6400183cddb58 Merge pull request #14619 from briansull/emitter-cleanup
dbfd10c94055fb5af823b4a7e30774641afcb3b2 Delete dead code (#14673)
1a0e2ef9c559af14acea429aecc591e1550b57db Merge pull request #14662 from wtgodbe/FeedTaskPackage
b813321f918b91142e95bc6b74d524664ea703ba Use new feed tasks package & auto-update version
4419496eb27ccc705c19c79ab4df714c06f507c3 Ifdef out legacy RegTracker code
b5a2484a9c1dbc751f77fe1934149c78d7f0ffc1 Cleanup unused emitter arguments Removed unused idClsCookie from struct instrDescDebugInfo Cleaned up several ifdefs
15e1472aab7ed42dafde0331ed6cece465a0763d Update BuildTools, CoreClr, CoreFx to prerelease-02120-09, preview1-25822-02, preview1-25824-01, respectively (#14565)
8c74b3bcd691d842a891bc55909a18a8b1e53200 Search/replace _DEBUG => DEBUG
```

**Comment by CarolEidt** [comment]
This was the case of a `BITCAST` above a `FIELD_LIST`, and was fixed by #14753
